code/modules/client/client_procs.dm

```diff
@@ -371,70 +372,73 @@ GLOBAL_LIST(external_rsc_urls)
+	var/sql_ip = sanitizeSQL(address)
+	var/sql_computerid = sanitizeSQL(computer_id)
 	var/sql_admin_rank = sanitizeSQL(admin_rank)
-
-
-	var/datum/DBQuery/query_log_player = SSdbcore.NewQuery("INSERT INTO [format_table_name("player")] (id, ckey, firstseen, lastseen, ip, computerid, lastadminrank) VALUES (null, '[sql_ckey]', Now(), Now(), INET_ATON('[sql_ip]'), '[sql_computerid]', '[sql_admin_rank]') ON DUPLICATE KEY UPDATE lastseen = VALUES(lastseen), ip = VALUES(ip), computerid = VALUES(computerid), lastadminrank = VALUES(lastadminrank)")
-	if(!query_log_player.Execute())
+	var/new_player
+	var/datum/DBQuery/query_client_in_db = SSdbcore.NewQuery("SELECT 1 FROM [format_table_name("player")] WHERE ckey = '[sql_ckey]'")
+	if(!query_client_in_db.Execute())
 		return
-
-	//Logging player access
-
+	if(!query_client_in_db.NextRow())
+		new_player = 1
+		account_join_date = sanitizeSQL(findJoinDate())
+		var/datum/DBQuery/query_add_player = SSdbcore.NewQuery("INSERT INTO [format_table_name("player")] VALUES ('[sql_ckey]', Now(), Now(), INET_ATON('[sql_ip]'), '[sql_computerid]', '[sql_admin_rank]', [account_join_date ? "'[account_join_date]'" : "NULL"])")
+		if(!query_add_player.Execute())
+			return
+		if(!account_join_date)
+			account_join_date = "Error"
+			account_age = -1
+	var/datum/DBQuery/query_get_client_age = SSdbcore.NewQuery("SELECT DATEDIFF(Now(),firstseen), DATEDIFF(Now(),accountjoindate) FROM [format_table_name("player")] WHERE ckey = '[sql_ckey]'")
+	if(!query_get_client_age.Execute())
+		return
+	if(query_get_client_age.NextRow())
+		player_age = text2num(query_get_client_age.item[1])
+		if(!account_join_date)
+			account_age = query_get_client_age.item[2]
+			if(!account_age)
+				account_join_date = sanitizeSQL(findJoinDate())
+				if(!account_join_date)
+					account_age = -1
+				else
+					var/datum/DBQuery/query_datediff = SSdbcore.NewQuery("SELECT DATEDIFF(Now(),accountjoindate)")
+					if(!query_datediff.Execute())
+						return
+					if(query_datediff.NextRow())
+						account_age = text2num(query_datediff.item[1])
+	if(!new_player)
+		var/datum/DBQuery/query_log_player = SSdbcore.NewQuery("UPDATE [format_table_name("player")] SET lastseen = Now(), ip = INET_ATON('[sql_ip]'), computerid = '[sql_computerid]', lastadminrank = '[sql_admin_rank]', accountjoindate = [account_join_date ? "'[account_join_date]'" : "NULL"] WHERE ckey = '[sql_ckey]'")
+		if(!query_log_player.Execute())
+			return
+	if(!account_join_date)
+		account_join_date = "Error"
 	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery("INSERT INTO `[format_table_name("connection_log")]` (`id`,`datetime`,`server_ip`,`server_port`,`ckey`,`ip`,`computerid`) VALUES(null,Now(),INET_ATON('[world.internet_address]'),'[world.port]','[sql_ckey]',INET_ATON('[sql_ip]'),'[sql_computerid]')")
 	query_log_connection.Execute()
 
+/client/proc/findJoinDate()
+	var/http[] = world.Export("http://byond.com/members/[ckey]?format=text")
```

Comment:
Well byond has no difference between [] and /list, it's just that we don't use this type of declaration anymore.